subprojects/core/src/main/java/org/gradle/internal/vfs/AbstractEventDrivenFileWatcherRegistry.java

```diff
@@ -93,8 +93,12 @@ private static Type convertType(FileWatcherCallback.Type type) {
     }
 
     @Override
-    public FileWatchingStatistics getStatistics() {
-        return new FileWatchingStatistics(unknownEventEncountered.get(), numberOfReceivedEvents.get(), errorWhileReceivingFileChanges.get());
+    public FileWatchingStatistics getAndResetStatistics() {
+        return new FileWatchingStatistics(
+            unknownEventEncountered.getAndSet(false),
+            numberOfReceivedEvents.getAndSet(0),
+            errorWhileReceivingFileChanges.getAndSet(null)
+        );
```

Comment:
I think we can improve the readability/safety here. The act of resetting the values and getting them at the same time is what makes this more complicated than necessary I think. Can we maybe have a way to replace the whole object, so we don't need to reset things, but can just start from scratch?

Reply:
The statics are now all part of one object.